packages/dev/core/src/Particles/particleSystem.ts

```diff
@@ -91,6 +92,42 @@ export class ParticleSystem extends ThinParticleSystem {
         return particleEmitter;
     }
 
+    /** Flow map */
+    private _flowMap: Nullable<FlowMap> = null;
+    private _flowMapUpdate: Nullable<_IExecutionQueueItem> = null;
+
+    /**
+     * The strength of the flow map
+     */
+    public flowMapStrength = 1.0;
+
+    /** Gets or sets the current flow map */
+    public get flowMap(): Nullable<FlowMap> {
+        return this._flowMap;
+    }
+
+    public set flowMap(value: Nullable<FlowMap>) {
+        if (this._flowMap === value) {
+            return;
+        }
+
+        this._flowMap = value;
+
+        if (value) {
+            this._flowMapUpdate = {
+                process: (particle: Particle) => {
+                    this._flowMap!._processParticle(particle, this, this.flowMapStrength);
+                },
+                previousItem: null,
+                nextItem: null,
+            };
+            _ConnectAfter(this._flowMapUpdate, this._directionProcessing!);
+        } else {
+            _RemoveFromQueue(this._flowMapUpdate!);
+        }

```

Comment:
`this._flowMapUpdate` should be removed from the queue even if `value !== null`.

Maybe:

```suggestion
        if (this._flowMapUpdate) {
            _RemoveFromQueue(this._flowMapUpdate);
            this._flowMapUpdate = null;
        }
        if (value) {
            this._flowMapUpdate = {
                process: (particle: Particle) => {
                    this._flowMap!._processParticle(particle, this, this.flowMapStrength);
                },
                previousItem: null,
                nextItem: null,
            };
            _ConnectAfter(this._flowMapUpdate, this._directionProcessing!);
        }
```

Reply:
nice catch!